### PR TITLE
Enable `remember` usage with `xs.create()`

### LINKED ISSUE #69: Remembered stream does not add listener when source is created directly via 'xs.create'
I noticed this odd behavior (what looks like a bug).  When `remember` is called directly on a raw stream created via `xs.create` that has no producer (i.e. which will emit based on the `shamefully*` methods), it doesn't add listeners since the `_prod` property is `undefined`.  
```
import xs from 'xstream'

const test$ = xs.create()

// This line works
//const oddBehavior$ = test$  

// This line does not work. Why?
const oddBehavior$ = test$.remember()

oddBehavior$.addListener({
  next: n => bin.log(`Next: ${n}`),
  error: e => bin.log(`Error: ${e}`),
  complete: () => bin.log(`Complete`)
})

test$.shamefullySendNext(1)
```
You can see the issue here: http://www.webpackbin.com/4JFCBWrL-

### PR DESCRIPTION
**Pull request by Hypnosphi**
Fixes #69

### CODE REVIEW

File: tests/operator/remember.ts [hunk 1]
```diff
@@ -131,4 +131,24 @@ describe('Stream.prototype.remember', () => {
     });
     done();
   });
+
+  it ('should work properly with `xs.create()`', (done) => {
+    const stream = xs.create().remember();
```
Comment (ntilwalli) [inline tests/operator/remember.ts]: This should be changed to create a reference to the `xs.create` stream and another references to the external `remember` stream.  The `shamefullySendNext` should be on the internal stream.  So it should read `const stream = xs.create(); const external = stream.remember();`

File: tests/operator/remember.ts [hunk 2]
```diff
@@ -131,4 +131,24 @@ describe('Stream.prototype.remember', () => {
...
+    stream.shamefullySendNext('foo');
+
+    let expected = ['foo', 'bar'];
+    stream.addListener({
```
Comment (ntilwalli) [inline tests/operator/remember.ts]: The should be changed to `external.addListener({`

**Comment by staltz** [comment]
Looks good, just need to fix the issues Nikhil mentioned.

**Comment by Hypnosphi** [comment]
Thank you, fixed this. Had to add one more listener to start the operator chain up.

File: tests/operator/remember.ts [hunk 3]
```diff
@@ -131,4 +131,27 @@ describe('Stream.prototype.remember', () => {
     });
     done();
   });
+
+  it ('should work properly with `xs.create()`', (done) => {
```
Comment (staltz) [inline tests/operator/remember.ts]: Can you split this into 2 tests? One test that has `shamefullySendNext('foo')` before `remembered.addListener` and another that has every shamefullySendNext after the addListener?

**Comment by Hypnosphi** [comment]
Oops, seems I'm doing something wrong with git

**Comment by Hypnosphi** [comment]
Splitting done, anyway

**Comment by staltz** [comment]
Thanks! Merged and will release in the next patch version soon

**Comment by ntilwalli** [comment]
@Hypnosphi @staltz Very cool. Thanks!

**Pull request merged**